site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,152 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import FormHelperText from "@mui/material/FormHelperText";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TextField, { type TextFieldProps } from "@mui/material/TextField";
+import { useState, type FC, useEffect } from "react";
+import {
+  type TimeUnit,
+  durationInDays,
+  durationInHours,
+  suggestedTimeUnit,
+} from "utils/time";
+
+type DurationFieldProps = Omit<TextFieldProps, "value" | "onChange"> & {
+  // Value is in ms
+  value: number;
+  onChange: (value: number) => void;
+};
+
+type State = {
+  unit: TimeUnit;
+  // Handling empty values as strings in the input simplifies the process,
+  // especially when a user clears the input field.
+  durationFieldValue: string;
+};
+
+export const DurationField: FC<DurationFieldProps> = (props) => {
+  const { value: parentValue, onChange, helperText, ...textFieldProps } = props;
+  const [state, setState] = useState<State>(() => initState(parentValue));
+  const currentDurationInMs = durationInMs(
+    state.durationFieldValue,
+    state.unit,
+  );
+
+  useEffect(() => {
+    if (parentValue !== currentDurationInMs) {
+      setState(initState(parentValue));
+    }
+  }, [currentDurationInMs, parentValue]);
+
+  return (
+    <div>
+      <div
+        css={{
+          display: "flex",
+          gap: 8,
+        }}
+      >
+        <TextField
+          {...textFieldProps}
+          type="number"
+          css={{ maxWidth: 160 }}
+          value={state.durationFieldValue}
+          onChange={(e) => {
+            const durationFieldValue = e.currentTarget.value;
+
+            setState((state) => ({
+              ...state,
+              durationFieldValue,
+            }));
+
+            const newDurationInMs = durationInMs(
+              durationFieldValue,
+              state.unit,
+            );
+            if (newDurationInMs !== parentValue) {
+              onChange(newDurationInMs);
+            }
+          }}
+          inputProps={{
+            step: 1,
+          }}
```

Comment:
Do we also want min and max values here? I feel like the `min` prop is probably safe to set to `0`, but I'm not sure if it's worth trying to put a max cap on the value, or if we can trust the server will take care of that

Reply:
Server and Yup, on the FE side, will take care of that. I'm only using step 1 here because it affects the UX since the user can click on the input arrows. Unfortunately, max only works for HTML validation 😞